docs/manual/howto/git.rst

```diff
@@ -0,0 +1,104 @@
+.. _using-git:
+
+=============
+Using ``git``
+=============
+
+``git`` is the version control system that is used to manage all of the source
+code. It is very powerful, but might be frightening at first.
+This page should give you a quick overview, but for a complete guide you will
+have to search the web on your own.
+Another great resource to get started practically without having to try out the
+newly-learned commands on a pre-existing repository is
+`learn git branching <https://learngitbranching.js.org>`_.
+You should probably learn the basic ``git`` vocabulary and then come back to
+find out how you can use all that practically. This guide will be oriented on
+how to create a pull request and things might be in a different order compared
+to the introductory guides.
+
+.. warning:: This guide is not complete and never will be. If something isn't
+   clear, consult other sources until you are confident you know what you are
+   doing.
+
+I want to try out a feature somebody is working on
+==================================================
+If you see a pull request on `GitHub <https://www.github.com/qtile/qtile/pulls>`_
+that you want to try out, have a look at the line where it says::
+
+  user wants to merge n commits into qtile:master from user:branch
+
+Right now you probably have one *remote* from which you can fetch changes, the
+``origin``. If you cloned ``qtile/qtile``, ``git remote show origin`` will spit
+out the *upstream* url. If you cloned your fork, ``origin`` points to it and you
+probably want to ``git remote add upstream https://www.github.com/qtile/qtile``.
+To try out somebody's work, you can add his fork as a new remote::
+
+  git remote add user https://www.github.com/user/qtile
+
+where you fill in the username from the line we asked you to search for before.
+Then you can load data from that remote with ``git fetch`` and then ultimately
+check out the branch with ``git checkout user/branch``.
+
+.. note:: Having the feature branch checked out doesn't mean that it is
+   installed and will be loaded when you restart qtile. You might still need to
+   install it with ``pip``.
+
+I committed changes and the tests failed
+========================================
+
+You can easily change your last commit: After you have done your work,
+``git add`` everything you need and use ``git commit --amend`` to change your
+last commit. This causes the git history of your local clone to be diverged from
+your fork on GitHub, so you need to force-push your changes with::
+
+  git push -f your_username
```

Comment:
I find this confusing as I never include my username when pushing because it's set via `git config` so I end up doing `git push -f origin <branch name>`.

Reply:
Actually, the username is the origin, i.e. I've set origin and upstream to qtile/qtile I think. Yeah, should definitely add an explanation to this...